proj/src/main/webapp/signup.js

```diff
@@ -143,15 +164,35 @@ function initializePasswordValidation(){
             passconf.classList.add("non-matching-pass");
             noMatch.hidden = false;
         }
-        if (meetRequirements(passVal) && noMatch.hidden) {
-            btnSignUp.disabled = false;
-        } else {
-            // disable sign up button
-            btnSignUp.disabled = true;
-        }
+        enableSignUp(passVal);
     })
 }
 
+function hasTag(){
+    const tagBox = document.getElementById("tags");
+    if (tagBox.value.trim().length != 0){
+        $('#tags').css('border','none');
+        return true;
+    } else {
+        // the tag box will only be in red after the user has input a valid password 
+        // this is to prevent the page from being visually overwhelming and help guide the user
+        if (noMatch.hidden){
+            $('#tags').css('border','3px solid #fa8072');
+        }
+        return false;
+    }
+}
+
+function enableSignUp(passVal){
+    $('#tags').keyup(hasTag);
+    if (meetRequirements(passVal) && noMatch.hidden && hasTag()){
```

Comment:
If you remove the line above as suggested in the above comment, hasTags will only get called if `noMatch.hidden` is true AND `meetsRequirements(passVal)` is true. So if hasTags has added the popup and then you go back and make the password invalid, then adding tags will not get rid of the pop up.

The reason is because `&&` shortcircuits  as soon as one of the values is false. If you use `&` then all three will still be called no matter if the first ones are true or false. If you use & instead of && make sure to leave a comment explaining why because it is uncommon to see and someone might come along later and think it is ok to add the extra &.

Reply:
ooh, thank you. i didn't know about `&`. i will change that right now